Sources/PlaydateKit/Core/Sound.swift

```diff
@@ -44,6 +50,37 @@ public enum Sound {
         @discardableResult public func play(repeat: Int32 = 1) -> Int32 {
             fileplayer.play.unsafelyUnwrapped(pointer, `repeat`)
         }
+        
+        private struct CallbackData {
+            var callback: (() -> Void)?
+        }
+        private var _callbackData: UnsafeMutablePointer<CallbackData>?
+
+        /// Installs a closure that will be called when playback has completed. This is implemented
+        /// by wrapping the closure in a struct that is stored in a dynamically allocated block, which
+        /// is passed as `userData` to `setFinishCallback` below.  
+        public var finishCallback: (() -> Void)? {
+            get {
+                _callbackData?.pointee.callback
+            }
+            set {
+                if let callbackData = _callbackData {
+                    callbackData.pointee.callback = newValue
+                } else if let newValue {
+                    let callbackData: UnsafeMutablePointer<CallbackData> = .allocate(capacity: 1)
+                    callbackData.initialize(to: .init(callback: newValue))
+                    _callbackData = callbackData
+                    setFinishCallback(
+                        callback: { sourceSource, userdata in
+                            if let callback = userdata?.assumingMemoryBound(to: CallbackData.self).pointee.callback {
+                                callback()
+                            }
+                        },
+                        soundUserdata: callbackData
+                    )
+                }
+            }
+        }
 
         /// Sets a function to be called when playback has completed.
         public func setFinishCallback(
```

Comment:
This should be marked as internal instead of public now, trying to hide all C wrappers once they have a swiftier version so you don't end up in weird states if you use both `setFinishCallback` and `finishCallback`

Reply:
See [1f2d7b4](https://github.com/finnvoor/PlaydateKit/pull/46/commits/1f2d7b4b8597c07d1613f2df00de06b073c03770)